fame/core/analysis.py

```diff
@@ -182,6 +184,40 @@ def add_ioc(self, value, source, tags=[]):
         self.collection.update_one({'_id': self['_id'], 'iocs.value': value},
                                    {'$addToSet': {'iocs.$.sources': source}})
 
+    def _store_preloaded_file(self, filepath=None, fd=None):
+        if fame_config.remote:
+            if not filepath and not fd:
+                raise ValueError(
+                    "Please provide either the path to the file or a file-like "
+                    "object containing the data.")
+
+            if filepath and fd:
+                self.log("debug",
+                        "Please provide either the path to the file or a "
+                        "file-like object containing the data, not both. "
+                        "Chosing the file-like object for now.")
+                response = send_file_to_remote(fd, '/files/')
+            else:
+                response = send_file_to_remote(filepath, '/files/')
+
+            return File(json_util.loads(response.text)['file'])
+        else:
+            return File(filename=os.path.basename(filepath), stream=fd)
+
+    def add_preloaded_file(self, filepath, fd):
+        f = self._store_preloaded_file(filepath, fd)
+        f.append_to('analysis', self['_id'])
+
+        if f['names'] == ['file']:
+            f['names'] = self._file['names']
+            f.save()
+
+        self['file'] = f['_id']
+        self._file = f
+        self.save()
+
+        self._automatic(preloading_done=True)
```

Comment:
If I understand correctly, this is creating a new File object with the actual file. I would expect the "hash" file to be replaced instead.

Reply:
Under the hood, this is exactly what we tried to implement. But as far as I remember, we could not (easily) update the currently used `File` object and opted for just creating a new one and replacing it in the `Analysis` object. I can check in the code if I now see a proper way to implement that - but I would like to postpone that until we are done with all the (upcoming) PRs.